brion/plugin/compartmentReportHDF5.cpp

```diff
@@ -302,45 +259,43 @@ void CompartmentReportHDF5::updateMapping(const GIDSet& gids)
 
     size_t nextCompartmentIndex = 0;
     size_t cellIndex = 0;
-    for (GIDSetCIter cellID = _gids.begin(); cellID != _gids.end();
-         ++cellID, ++cellIndex)
+    for (auto cellID : _gids)
     {
-        _openFile(*cellID);
+        _openFile(cellID);
 
-        const H5::H5File& file =
-            _file.getId() ? _file : _files.find(*cellID)->second;
+        const HighFive::File& file =
+            _file ? *_file : _files.find(cellID)->second;
 
         std::stringstream cellName;
-        cellName << "a" << *cellID;
+        cellName << "a" << cellID;
         const std::string datasetName =
             "/" + cellName.str() + "/" + _reportName + "/" + mappingDatasetName;
-        H5::DataSet dataset;
-        H5E_BEGIN_TRY
-        dataset = file.openDataSet(datasetName);
-        H5E_END_TRY
-        if (!dataset.getId())
-        {
-            LBTHROW(
-                std::runtime_error("ReportReaderHDF5: "
-                                   "Dataset " +
-                                   datasetName + " not found "
-                                                 "in file: " +
-                                   file.getFileName()));
-        }
+        HighFive::DataSet dataset = [file, datasetName]() {
+            try
+            {
+                return file.getDataSet(datasetName);
+            }
+            catch (HighFive::DataSetException&)
+            {
+                LBTHROW(std::runtime_error("CompartmentReportHDF5: Dataset " +
+                                           datasetName + " not found "
+                                                         "in file: " +
+                                           file.getName()));
+            }
+        }();
 
-        if (dataset.getSpace().getSimpleExtentNdims() != 2)
+        auto dims = dataset.getSpace().getDimensions();
+        if (dims.size() != 2)
         {
             LBTHROW(
                 std::runtime_error("Compartment_Report_HDF5_File_Reader: "
                                    "Error, not 2 dimensional array on " +
                                    datasetName));
         }
 
-        hsize_t dims[2];
-        dataset.getSpace().getSimpleExtentDims(dims);
-
         boost::scoped_array<float> buffer(new float[dims[1]]);
-        dataset.read(buffer.get(), H5::PredType::NATIVE_FLOAT);
+        float* ptr = buffer.get(); // HighFive is picky with argument passing
```

Comment:
that comment does not make sense to me, maybe I don't know too much about highfive. What do you mean here?

Reply:
I'll rephrase it. It means that HighFive expects an l-value for the output buffer (because it could be a std::vector passed by value) so I need to create a temporary variable to call read.